### ISSUE TITLE
Unexpected outcome parsing `@param array{stdClass|null}`

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#328]
I assume it's due to `:` being expected after stdClass?

Observed: `InvalidArrayOffset: ... Cannot access value on variable $args using int offset, expecting 'stdClass'`
Expected: The intdiv should warn about getting stdClass (or null)

```php
<?php

/**
 * @param array{stdClass|null} $args
 */
function test2($args) : void {
    $obj = $args[0];
    echo intdiv($obj, 2); 
}
```

**Comment by muglug** [comment]
ah, parsing of `array{stdClass|null}` isn't too hot. Goes away if you switch to `array{0:stdClass|null}`
